### PR TITLE
Extract hard-coded root folder.

### PR DESCRIPTION
**Pull request by leslielazzarino**
Signed-off-by: Leslie Lazzarino <leslie.lazzarino@tngtech.com>

### Summary of changes

The resource root folder, added in the UI to the resources provided in the input file, was hardcoded inside the VirtualizedTree. Now it added to the resources before giving them to the tree. 

### Context and reason for change

Step in the direction of making the VirtualizedTree more generic and extract it to a separate package

### How can the changes be tested

Basically just a refactoring. The file tree in the app should still look and work as before.

Note: Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

### CODE REVIEW

File: src/Frontend/Components/ResourceBrowser/ResourceBrowser.tsx [hunk 1]
```diff
@@ -97,16 +97,16 @@ export function ResourceBrowser(): ReactElement | null {
...
 
-  return (
+  return resources ? (
     <VirtualizedTree
       expandedIds={expandedIds}
       isFileWithChildren={getFileWithChildrenCheck(filesWithChildren)}
       onSelect={handleSelect}
       onToggle={handleToggle}
-      resources={resources}
+      resources={{ '': resources }}
```
Comment (nicarl) [inline src/Frontend/Components/ResourceBrowser/ResourceBrowser.tsx:106]: Can you move this to a constant? 
```typescript
const ROOT_FOLDER = "";

File: src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx [hunk 2]
```diff
@@ -75,16 +75,24 @@ export function VirtualizedTree(
...
+              props.onSelect,
+              props.onToggle,
+              props.getTreeItemLabel
+            )
+          );
+          return treeItems;
+        },
+        []
       )
     : [];
```
Comment (nicarl) [inline src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx:97]: If `VirtualizedTree` becomes a "public" component I am not sure if resources should be nullable.
I would expect.
```typescript
resources: Resources;
```

File: src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx [hunk 3]
```diff
@@ -75,16 +75,24 @@ export function VirtualizedTree(
...
-        classes,
-        props.expandedIds,
-        props.selectedResourceId,
-        props.isFileWithChildren,
-        props.onSelect,
-        props.onToggle,
-        props.getTreeItemLabel
+    ? Object.keys(props.resources).reduce(
+        (treeItems: Array<ReactElement>, resourceName: string) => {
+          treeItems = treeItems.concat(
```
Comment (nicarl) [inline src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx:80]: A note for the future:
This could become a bottleneck for huge trees. `concat` returns a new array instead of modifying the existing `treeItems'. Before publishing the tree on npm it would be amazing to test the component with a really large tree.

**Review by nicarl — APPROVED** [review]

File: src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx [hunk 4]
```diff
@@ -75,16 +75,24 @@ export function VirtualizedTree(
...
-        { '': props.resources },
-        '',
-        classes,
-        props.expandedIds,
-        props.selectedResourceId,
-        props.isFileWithChildren,
-        props.onSelect,
-        props.onToggle,
-        props.getTreeItemLabel
+    ? Object.keys(props.resources).reduce(
```
Comment (benedikt-richter) [inline src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx:78]: should be tested

**Review by benedikt-richter — CHANGES_REQUESTED** [review]
see comment

File: src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx (hunk 4 above)
Comment (leslielazzarino) [reply src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx]: Turns out it wasn't needed.

File: src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx (hunk 3 above)
Comment (leslielazzarino) [reply src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx]: Turns out it wasn't needed.

**Review by benedikt-richter — APPROVED** [review]
